### PR TITLE
Add option to disable Undo Commit popup

### LINKED ISSUE #15134: Impossibile to deactivate undo prompt
### The problem

I disabled all prompts under settings, but apparently there is no setting to deactivate the UNDO commit prompt.
Screenshot 2022-08-18 at 13 14 57

Screenshot 2022-08-18 at 13 16 59

### Release version

Version 3.0.5 (x64)

### Operating system

macOS

### Steps to reproduce the behavior

1. create a new commit
2. press "UNDO"
3. a prompt is displayed

### Log files

_No response_

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by geophilusd**
Closes #15134 

## Description
- Added an option in prompts to disable the Undo Commit prompt
- Added a checkbox "Do not show this message again" in the Undo Commit Confirmation popup
- Undo Commit Confirmation box will appear for Merge commit irrespective of the setting in Preferences -> Prompt

### Screenshots
Screenshot 2022-09-07 at 9 04 35 AM
Screenshot 2022-09-07 at 9 05 34 AM

## Release notes
Notes:
Add an option to disable the Undo Commit popup

### CODE REVIEW

File: app/src/lib/stores/app-store.ts [hunk 1]
```diff
@@ -5190,6 +5200,14 @@ export class AppStore extends TypedBaseStore<IAppState> {
...
   }
 
+  public _setConfirmUndoCommitSetting(value: boolean): Promise<void> {
+    this.confirmUndoCommit = value
+    setBoolean(confirmForcePushKey, value)
+
+    this.emitUpdate()
+
+    return Promise.resolve()
+  }
```
Comment (sergiou87) [inline app/src/lib/stores/app-store.ts:5212]: Nit: add blank line after this:
```suggestion
  }

```

File: app/src/lib/stores/app-store.ts [hunk 2]
```diff
@@ -4551,7 +4560,8 @@ export class AppStore extends TypedBaseStore<IAppState> {
     // Warn the user if there are changes in the working directory
```
Comment (sergiou87) [inline app/src/lib/stores/app-store.ts:4560]: Could we add another comment here to explain why this condition? Something like…
```suggestion
    // Warn the user if there are changes in the working directory
    // This warning can be disabled, except when the user tries to undo
    // a merge commit.
```

File: app/src/ui/undo/warn-local-changes-before-undo.tsx [hunk 3]
```diff
@@ -44,48 +50,66 @@ export class WarnLocalChangesBeforeUndo extends React.Component<
...
+                  : CheckboxValue.On
+              }
+              onChange={this.onConfirmUndoCommitChanged}
+            />
+          </Row>
+        </DialogContent>
       )
-    } else if (this.props.commit.isMergeCommit) {
+    }
+  }
```
Comment (sergiou87) [inline app/src/ui/undo/warn-local-changes-before-undo.tsx]: You can get rid of the `else` and just leave the `if` as a "fast return":
```typescript
    if (this.props.commit.isMergeCommit) {
      return this.getMergeCommitWarningDialog()
    }

    return ...
```

File: app/src/ui/undo/warn-local-changes-before-undo.tsx [hunk 4]
```diff
@@ -44,48 +50,66 @@ export class WarnLocalChangesBeforeUndo extends React.Component<
...
+        <DialogContent>
+          <Row>
+            {this.getMergeCommitUndoWarningText()}
+            <br />
+            <br />
+            Do you want to continue anyway?
+          </Row>
+        </DialogContent>
       )
     } else {
```
Comment (sergiou87) [inline app/src/ui/undo/warn-local-changes-before-undo.tsx]: Same about this `else`

**Review by sergiou87 — CHANGES_REQUESTED** [review]
This is great! Thank you for your contribution 🙇‍♂️ 

It all looks good, I just found a few minor style things that could be improved. I'll test it after they're solved 😄

File: app/src/lib/stores/app-store.ts (hunk 2 above)
Comment (geophilusd) [reply app/src/lib/stores/app-store.ts:4560]: done

File: app/src/lib/stores/app-store.ts (hunk 1 above)
Comment (geophilusd) [reply app/src/lib/stores/app-store.ts:5212]: done

File: app/src/ui/undo/warn-local-changes-before-undo.tsx (hunk 3 above)
Comment (geophilusd) [reply app/src/ui/undo/warn-local-changes-before-undo.tsx]: refactored as you suggested

File: app/src/ui/undo/warn-local-changes-before-undo.tsx (hunk 4 above)
Comment (geophilusd) [reply app/src/ui/undo/warn-local-changes-before-undo.tsx]: done

**Comment by geophilusd** [comment]
@sergiou87 made the requested changes. Please check

**Review by sergiou87 — APPROVED** [review]
Thank you for those changes! Works perfectly well too 🙇‍♂️

**Pull request merged**